src/server/api/browse.ts

```diff
@@ -9,39 +9,50 @@ import type {
   DiscoverySubjectsResponse,
 } from "../../types/browseTypes"
 import { getBrowseQuery } from "../../utils/browseUtils"
+import type { APIError } from "../../types/appTypes"
 
 export async function fetchSubjects(
   browseParams?: BrowseParams
-): Promise<DiscoverySubjectsResponse | { status: number; message?: string }> {
+): Promise<DiscoverySubjectsResponse | APIError> {
   const browseQuery = getBrowseQuery(browseParams)
 
   try {
     // Failure to build client will throw from this:
     const client = await nyplApiClient()
-    const res = await client.get(
+    const results = await client.get(
       `${DISCOVERY_API_BROWSE_ROUTE}${browseQuery}&per_page=${SUBJECTS_PER_PAGE.toString()}`
     )
-    // Handle empty results (404)
-    if (res.status === 404 || res.subjects.length === 0) {
+
+    // Handle no results (404)
+    if (results?.totalResults === 0) {
+      return {
+        status: 404,
+        error: `No results found for ${DISCOVERY_API_BROWSE_ROUTE}${browseQuery}&per_page=${SUBJECTS_PER_PAGE.toString()}`,
```

Comment:
Do we have access to the original request url here? if so, i think it will be safer not to reconstruct urls in the logs

Reply:
I'd recommend setting the value in client.get() to a variable and use that to populate the logs in this file and the other server fetch methods